forest/daemon/src/daemon.rs

```diff
@@ -392,6 +394,22 @@ pub(super) async fn start(config: Config, detached: bool) -> Db {
     db
 }
 
+async fn propagate_error(services: &mut JoinSet<Result<(), anyhow::Error>>) -> anyhow::Error {
+    while !services.is_empty() {
+        select! {
+            option = services.join_next().fuse() => {
+                if let Some(Ok(Err(error_message))) = option {
+                    return error_message
+                }
+            },
+        }
+    }
+    tokio::time::sleep(Duration::new(3200000000, 0)).await;
```

Comment:
Does `tokio::time::sleep` have a max sleeping duration? What happens if you exceed the max duration?

Reply:
Approximately 2.2 years. I fixed our time to be within the limit

Reply:
I don't see your fix. Please push your changes before marking issues as resolved. That makes it much easier for reviewers to keep track of things.